src/exercise1/final/BookClient/BookClient/MainPage.xaml.cs

```diff
@@ -26,13 +23,29 @@ public MainPage()
 
         async void OnRefresh(object sender, EventArgs e)
         {
-            var bookCollection = await manager.GetAll();
+            if (IsBusy)
+                return;
 
-            foreach (Book book in bookCollection)
+            try
             {
-                if (books.All(b => b.ISBN != book.ISBN))
-                    books.Add(book);
+                IsBusy = true;
+                var bookCollection = await manager.GetAll();
+                foreach (Book book in bookCollection)
+                {
+                    if (books.All(b => b.ISBN != book.ISBN))
+                        books.Add(book);
+                }
             }
+            catch (Exception ex)
+            {
+                await this.DisplayAlert("Error",
+                        ex.Message,
+                        "OK");
+            }
+            finally
+            {
+                IsBusy = false;
+            }            
```

Comment:
Trimming trailing whitespace
```suggestion
            }
```